In [28]:
pip install pandas openpyxl

Note: you may need to restart the kernel to use updated packages.


In [29]:
pip install pyecharts

Note: you may need to restart the kernel to use updated packages.


In [30]:
from pathlib import Path  # Python Standard Library
import pandas as pd  # pip install pandas openpyxl
from pyecharts import options as opts  # pip install pyecharts
from pyecharts.charts import Bar, Calendar, Tab

In [31]:
# Read Excel file (Financial_Data.xlsx) into a pandas DataFrame.
df = pd.read_excel("Financial_Data of Coding is Fun Ltd..xlsx",
    engine= "openpyxl",
    sheet_name= "Orders",
    skiprows= 2,
    usecols= "B:T",
    nrows= 3312,
)
df.head()

Order ID Order Date       Ship Mode Customer ID    Customer Name  \
0  CA-2017-114412 2021-04-15  Standard Class    AA-10480     Andrew Allen   
1  US-2017-156909 2021-07-16    Second Class    SF-20065  Sandra Flanagan   
2  CA-2017-107727 2021-10-19    Second Class    MA-17560     Matt Abelman   
3  CA-2017-120999 2021-09-10  Standard Class    LC-16930   Linda Cazamias   
4  CA-2017-139619 2021-09-19  Standard Class    ES-14080       Erin Smith   

       Segment        Country          City           State  Postal Code  \
0     Consumer  United States       Concord  North Carolina        28027   
1     Consumer  United States  Philadelphia    Pennsylvania        19140   
2  Home Office  United States       Houston           Texas        77095   
3    Corporate  United States    Naperville        Illinois        60540   
4    Corporate  United States     Melbourne         Florida        32935   

    Region       Product ID         Category Sub-Category  \
0    South  OFF-PA-10002365  Office Supplies        Paper   
1     East  FUR-CH-10002774        Furniture       Chairs   
2  Central  OFF-PA-10000249  Office Supplies        Paper   
3  Central  TEC-PH-10004093       Technology       Phones   
4    South  OFF-ST-10003282  Office Supplies      Storage   

                                        Product Name    Sales  Quantity  \
0                                         Xerox 1967   15.552         3   
1                 Global Deluxe Stacking Chair, Gray   71.372         2   
2                                  Easy-staple paper   29.472         3   
3                                 Panasonic Kx-TS550  147.168         4   
4  Advantus 10-Drawer Portable Organizer, Chrome ...   95.616         2   

   Discount   Profit  
0       0.2   5.4432  
1       0.3  -1.0196  
2       0.2   9.9468  
3       0.2  16.5564  
4       0.2   9.5616

In [32]:
# Get a quick info about our dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3312 entries, 0 to 3311
Data columns (total 19 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Order ID       3312 non-null   object        
 1   Order Date     3312 non-null   datetime64[ns]
 2   Ship Mode      3312 non-null   object        
 3   Customer ID    3312 non-null   object        
 4   Customer Name  3312 non-null   object        
 5   Segment        3312 non-null   object        
 6   Country        3312 non-null   object        
 7   City           3312 non-null   object        
 8   State          3312 non-null   object        
 9   Postal Code    3312 non-null   int64         
 10  Region         3312 non-null   object        
 11  Product ID     3312 non-null   object        
 12  Category       3312 non-null   object        
 13  Sub-Category   3312 non-null   object        
 14  Product Name   3312 non-null   object        
 15  Sales          3312 n

In [33]:
# Add month column to our dataframe
df['Month'] = df['Order Date'].dt.month
df.head(3)

Order ID Order Date       Ship Mode Customer ID    Customer Name  \
0  CA-2017-114412 2021-04-15  Standard Class    AA-10480     Andrew Allen   
1  US-2017-156909 2021-07-16    Second Class    SF-20065  Sandra Flanagan   
2  CA-2017-107727 2021-10-19    Second Class    MA-17560     Matt Abelman   

       Segment        Country          City           State  Postal Code  \
0     Consumer  United States       Concord  North Carolina        28027   
1     Consumer  United States  Philadelphia    Pennsylvania        19140   
2  Home Office  United States       Houston           Texas        77095   

    Region       Product ID         Category Sub-Category  \
0    South  OFF-PA-10002365  Office Supplies        Paper   
1     East  FUR-CH-10002774        Furniture       Chairs   
2  Central  OFF-PA-10000249  Office Supplies        Paper   

                         Product Name   Sales  Quantity  Discount  Profit  \
0                          Xerox 1967  15.552         3       0.2  5.4432   
1  Global Deluxe Stacking Chair, Gray  71.372         2       0.3 -1.0196   
2                   Easy-staple paper  29.472         3       0.2  9.9468   

   Month  
0      4  
1      7  
2     10

In [34]:
import pandas as pd

# Assuming 'Order Date' is the name of the datetime column
df['Order Date'] = pd.to_datetime(df['Order Date'])
df['Month'] = df['Order Date'].dt.to_period("M")

# Group dataframe by months
grouped_by_months = df.groupby(by=['Month']).agg({'Sales': 'sum', 'Profit': 'sum'}).reset_index()
grouped_by_months.head(3)


Month       Sales      Profit
0  2021-01  43971.3740   7140.4391
1  2021-02  20301.1334   1613.8720
2  2021-03  58872.3528  14751.8915

In [35]:
# plot sales & profit by months
bar_chart_by_month = (
    Bar()
    .add_xaxis(grouped_by_months.index.tolist())
    .add_yaxis('Sales', grouped_by_months['Sales'].round(0).tolist())
    .add_yaxis('Profit', grouped_by_months['Profit'].round(0).tolist())
    .set_global_opts(
        title_opts=opts.TitleOpts(title='Sales & Profit by month', subtitle='in USD')
    )
)

In [36]:
bar_chart_by_month.render_notebook()

In [37]:
# Group data by sub-category
grouped_by_sub_category = df.groupby(by='Sub-Category', as_index=False).agg({'Sales': 'sum', 'Profit': 'sum'}).sort_values(by=['Sales'])
grouped_by_sub_category.head(3)


Sub-Category     Sales     Profit
8     Fasteners   857.594   304.9489
7     Envelopes  3378.574  1441.7590
10       Labels  3861.216  1744.6093

In [38]:
# plot sales & profit by subcategroy
bar_chart_by_subcategory = (
    Bar()
    .add_xaxis(grouped_by_sub_category['Sub-Category'].tolist())
    .add_yaxis('Sales', grouped_by_sub_category['Sales'].round(0).tolist())
    .add_yaxis('Profit', grouped_by_sub_category['Profit'].round(0).tolist())
    .reversal_axis()
    .set_series_opts(label_opts=opts.LabelOpts(position='right'))
    .set_global_opts(
        title_opts=opts.TitleOpts(title='Sales & Profit by subcategory', subtitle='in USD')
    )
)

In [39]:
bar_chart_by_subcategory.render_notebook()

In [40]:
# Group data by Order Date
sales_by_date = df.groupby(by='Order Date').agg({'Sales': 'sum'}).round(0).reset_index()
sales_by_date.head(3)


Order Date   Sales
0 2021-01-01  1482.0
1 2021-01-02  2080.0
2 2021-01-03  2070.0

In [41]:
data = sales_by_date[['Order Date', 'Sales']].values.tolist()
data

[[Timestamp('2021-01-01 00:00:00'), 1482.0],
 [Timestamp('2021-01-02 00:00:00'), 2080.0],
 [Timestamp('2021-01-03 00:00:00'), 2070.0],
 [Timestamp('2021-01-06 00:00:00'), 34.0],
 [Timestamp('2021-01-07 00:00:00'), 3396.0],
 [Timestamp('2021-01-08 00:00:00'), 893.0],
 [Timestamp('2021-01-09 00:00:00'), 274.0],
 [Timestamp('2021-01-12 00:00:00'), 849.0],
 [Timestamp('2021-01-13 00:00:00'), 4619.0],
 [Timestamp('2021-01-14 00:00:00'), 1056.0],
 [Timestamp('2021-01-15 00:00:00'), 2140.0],
 [Timestamp('2021-01-16 00:00:00'), 6230.0],
 [Timestamp('2021-01-19 00:00:00'), 2014.0],
 [Timestamp('2021-01-20 00:00:00'), 768.0],
 [Timestamp('2021-01-21 00:00:00'), 2547.0],
 [Timestamp('2021-01-22 00:00:00'), 4212.0],
 [Timestamp('2021-01-23 00:00:00'), 450.0],
 [Timestamp('2021-01-24 00:00:00'), 419.0],
 [Timestamp('2021-01-26 00:00:00'), 3055.0],
 [Timestamp('2021-01-27 00:00:00'), 332.0],
 [Timestamp('2021-01-28 00:00:00'), 1008.0],
 [Timestamp('2021-01-29 00:00:00'), 295.0],
 [Timestamp('2021-01

In [42]:
# calculate min / max sales
max_sales = df['Sales'].max()
min_sales = df['Sales'].min()

In [43]:
sales_calendar = (
    Calendar()
    .add('', data, calendar_opts=opts.CalendarOpts(range_='2021'))
    .set_global_opts(
        title_opts=opts.TitleOpts(title='Sales Calendar', subtitle='in USD'),
        legend_opts=opts.LegendOpts(is_show=False),
        visualmap_opts=opts.VisualMapOpts(
                max_=max_sales,
                min_=min_sales,
                orient='horizontal',
                is_piecewise=False,
                pos_top='230px',
                pos_left='100px',
        )
    )
)

In [44]:
sales_calendar.render_notebook()

In [45]:
import pandas as pd
import plotly.express as px

# Assuming your DataFrame has "Region" and "Sales" columns
fig = px.pie(df, names='Region', values='Sales', title='Sales Distribution by Region',
             labels={'Sales': 'Total Sales'}, 
             hover_data=['Region', 'Sales'],
             template='plotly',
             color_discrete_sequence=['Orange', 'blue', 'red', 'yellow'])

# Show the plot
fig.show()


In [46]:
import pandas as pd
import plotly.express as px

# Create an interactive plot using Plotly Express
fig = px.sunburst(df, path=['Region', 'Ship Mode'], title='Interactive Plot: Ship Mode and Region')

# Show the plot
fig.show()


In [47]:
import plotly.express as px
import pandas as pd

# Create an interactive box plot of quantity by region
fig = px.box(df, x='Region', y='Quantity',
             color='Region',  # Color by region for better differentiation
             labels={'Quantity': 'Total Quantity'},
             title='Interactive Box Plot: Quantity by Region',
             template='plotly_dark',  # Dark template for a stylish appearance
             points="all",  # Display all data points
             hover_data=['Sub-Category', 'Quantity'],  # Additional information to display on hover
             height=500
             )

# Show the plot
fig.show()


In [48]:
import plotly.express as px
import pandas as pd

# Create an interactive scatter plot of discount vs. profit
fig = px.scatter(df, x='Discount', y='Profit',
                 color='Sub-Category',
                 size='Sales',
                 labels={'Discount': 'Discount %', 'Profit': 'Total Profit'},
                 title='Interactive Scatter Plot: Discount vs. Profit',
                 template='plotly_dark',  # Dark template for a stylish appearance
                 hover_data=['Sub-Category', 'Sales'],
                 height=500
                 )

# Update marker style for a more appealing look
fig.update_traces(marker=dict(line=dict(color='white', width=1)),
                  selector=dict(mode='markers'))

# Update layout for better aesthetics
fig.update_layout(
    font=dict(family="Arial", size=12, color="White"),
    title=dict(font=dict(size=24)),
    xaxis=dict(title=dict(font=dict(size=18))),
    yaxis=dict(title=dict(font=dict(size=18))),
    legend=dict(title=dict(font=dict(size=16))),
)

# Show the plot
fig.show()


In [49]:
import dash
import dash_core_components as dcc
import dash_html_components as html
from dash.dependencies import Input, Output
import plotly.graph_objects as go
import plotly.express as px
import pandas as pd

# Assuming df is your DataFrame
# ...

df['Order Date'] = pd.to_datetime(df['Order Date'])
df['Month'] = df['Order Date'].dt.to_period("M")
grouped_by_months = df.groupby(by=['Month']).agg({'Sales': 'sum', 'Profit': 'sum'}).reset_index()

grouped_by_sub_category = df.groupby(by='Sub-Category', as_index=False).agg({'Sales': 'sum', 'Profit': 'sum'}).sort_values(by=['Sales'])

sales_by_date = df.groupby(by='Order Date').agg({'Sales': 'sum'}).round(0).reset_index()

data = sales_by_date[['Order Date', 'Sales']].values.tolist()
max_sales = df['Sales'].max()
min_sales = df['Sales'].min()

# Initialize the Dash app
app = dash.Dash(__name__)

# Define app layout
app.layout = html.Div(
    style={'backgroundColor': '#001F3F', 'color': 'white', 'padding': '10px'},
    children=[
        html.H1("Financial Dashboard - Coding is Fun Ltd.", style={'marginBottom': '20px'}),

        # Sales and Profit by Month
        dcc.Graph(
            id='bar_chart_by_month',
            figure={
                'data': [
                    go.Bar(x=grouped_by_months.index, y=grouped_by_months['Sales'].round(0), name='Sales'),
                    go.Bar(x=grouped_by_months.index, y=grouped_by_months['Profit'].round(0), name='Profit')
                ],
                'layout': go.Layout(
                    title='Sales & Profit by month',
                    barmode='stack',
                    xaxis=dict(title='Month'),
                    yaxis=dict(title='Amount in USD'),
                    plot_bgcolor='#000000',
                    paper_bgcolor='#000000',
                    font=dict(color='white')
                )
            }
        ),

        # Sales and Profit by Sub-Category
        dcc.Graph(
            id='bar_chart_by_subcategory',
            figure={
                'data': [
                    go.Bar(x=grouped_by_sub_category['Sub-Category'], y=grouped_by_sub_category['Sales'].round(0), name='Sales'),
                    go.Bar(x=grouped_by_sub_category['Sub-Category'], y=grouped_by_sub_category['Profit'].round(0), name='Profit')
                ],
                'layout': go.Layout(
                    title='Sales & Profit by subcategory',
                    barmode='stack',
                    xaxis=dict(title='Sub-Category'),
                    yaxis=dict(title='Amount in USD'),
                    plot_bgcolor='#000000',
                    paper_bgcolor='#000000',
                    font=dict(color='white')
                )
            }
        ),

        # Sales Calendar
        dcc.Graph(
            id='sales_calendar',
            figure={
                'data': [
                    go.Scatter(
                        x=sales_by_date['Order Date'],
                        y=sales_by_date['Sales'],
                        mode='markers',
                        marker=dict(size=8, color=sales_by_date['Sales'], colorscale='Viridis'),
                        text=sales_by_date['Sales'],
                        hoverinfo='text+x+y'
                    )
                ],
                'layout': go.Layout(
                    title='Sales Calendar',
                    xaxis=dict(title='Order Date'),
                    yaxis=dict(title='Sales Amount in USD'),
                    showlegend=False,
                    plot_bgcolor='#000000',
                    paper_bgcolor='#000000',
                    font=dict(color='white')
                )
            }
        ),

        # Pie chart
        dcc.Graph(
            id='sales_pie_chart',
            figure=px.pie(df, names='Region', values='Sales',
                          title='Sales Distribution by Region',
                          labels={'Sales': 'Total Sales'},
                          hover_data=['Region', 'Sales'],
                          template='plotly_dark',
                          color_discrete_sequence=['Orange', 'blue', 'red', 'yellow']
            )
        ),

        # Sunburst chart
        dcc.Graph(
            id='sunburst_chart',
            figure=px.sunburst(df, path=['Region', 'Ship Mode'], title='Interactive Plot: Ship Mode and Region').update_layout(
                plot_bgcolor='#000000',
                paper_bgcolor='#000000',
                font=dict(color='white')
            )
        ),

        # Box Plot: Quantity by Region
        dcc.Graph(
            id='box_plot_quantity_region',
            figure=px.box(df, x='Region', y='Quantity',
                          color='Region',
                          labels={'Quantity': 'Total Quantity'},
                          title='Interactive Box Plot: Quantity by Region',
                          template='plotly_dark',
                          points="all",
                          hover_data=['Sub-Category', 'Quantity'],
                          height=500
                          )
        ),

        # Scatter Plot: Discount vs. Profit
        dcc.Graph(
            id='scatter_discount_profit',
            figure=px.scatter(df, x='Discount', y='Profit',
                              color='Sub-Category',
                              size='Sales',
                              labels={'Discount': 'Discount %', 'Profit': 'Total Profit'},
                              title='Interactive Scatter Plot: Discount vs. Profit',
                              template='plotly_dark',
                              hover_data=['Sub-Category', 'Sales'],
                              height=500
                              ).update_traces(marker=dict(line=dict(color='white', width=1)),
                                             selector=dict(mode='markers')).update_layout(
                font=dict(family="Arial", size=12, color="White"),
                title=dict(font=dict(size=24, color='white')),
                xaxis=dict(title=dict(font=dict(size=18, color='white'))),
                yaxis=dict(title=dict(font=dict(size=18, color='white'))),
                legend=dict(title=dict(font=dict(size=16, color='white'))),
            )
        ),
    ]
)

if __name__ == '__main__':
    app.run_server(debug=True)


C:\Users\Parth narula\anaconda3\lib\site-packages\_plotly_utils\basevalidators.py:106: FutureWarning:

The behavior of DatetimeProperties.to_pydatetime is deprecated, in a future version this will return a Series containing python datetime objects instead of an ndarray. To retain the old behavior, call `np.array` on the result



# Project Report

# Objectives

The overall objective of this project report is to ascertain the Financial Report of Coding is Fun Ltd. which is a firm and to gauge on the potential insights we can derive through it's sales, profitability, and factors like regions were products are sold, mode of shipment etc to help the management team of the coding is Fun Ltd. firm to focus on some key areas on the basis of the interactive dashboard we are providing and the analytical tools we are generating which makes it easy to analyse things and take decisions in more efficent an deffective way.
Furthermore an indepth objective analysis on the report is as follows:

1. Sales and Profit Overview:
To Provide a comprehensive overview of total sales and profits apart from that, analyze the overall financial performance of Coding Is Fun Ltd.

2. Order Date Analysis:
To Explore sales trends over time based on the order date and Identify peak sales periods and potential seasonality.

3. Ship Mode Insights:
To Analyze the distribution of sales across different shipping modes and Identify the most commonly used shipping modes.

4. Regional Performance:
To Break down sales and profits by region to identify top-performing and underperforming regions and also to Facilitate regional-level decision-making.


5. Product Category and Sub-Category Analysis:
To Explore sales and profit distribution across product categories and sub-categories and to Identify the most lucrative and popular product categories.

6. Product-wise Performance:
To Highlight the top-selling and most profitable products and to Identify opportunities for promotion or inventory optimization.

7. Quantity and Discount Relationships:
To Analyze the relationship between the quantity of products sold, discounts offered, and overall profit.Apart from that,
understand the impact of discounts on sales and profitability.



# Briefs on the Data & the Analysis


The dataset consists of 3312 rows and 19 columns, representing order and financial information for Coding is fun Ltd.. The various aspects that the data brief covers are:

1. Order Information: The dataset includes an "Order ID" column, providing a unique identifier for each order. The "Order Date" column records the date when the order was placed, allowing for temporal analysis.

2. Customer Details: Customer-related information is captured through the "Customer ID" and "Customer Name" columns, offering unique identifiers and names for individual customers.

3. Geographic Information: The dataset contains geographical details such as "Country," "City," "State," "Postal Code," and "Region," allowing for the exploration of regional trends and customer distribution.

4. Product Information: Product-related data is comprehensive, with columns like "Product ID," "Category," "Sub-Category," and "Product Name," providing insights into the diversity and categorization of the company's product offerings.

5. Financial Metrics: The financial aspect is represented by columns like "Sales," "Quantity," "Discount," and "Profit." These metrics enable the analysis of revenue generation, sales quantities, discount impact, and overall profitability for each order.

6. Logistics: The "Ship Mode" column indicates the mode of shipment chosen for each order, offering insights into the company's logistical operations.

The dataset is structured and comprehensive, suitable for various analyses related to sales performance, customer behavior, and financial insights. Potential analyses could include examining trends in sales over time, understanding customer preferences, evaluating the impact of discounts on profitability, and optimizing logistics strategies. Overall, this dataset serves as a valuable resource for extracting meaningful insights into the company's business dynamics and guiding strategic decision-making.







# Analysis for Plot 1 (Bar Chart with Months)

#Objectives

Objective 1: Identify and leverage the factors contributing to high sales months, such as March, September, and November.

Objective 2: Improve profit margins across all months by understanding the correlation between high sales and high profits.

Objective 3: Capitalize on seasonal trends and events to drive both sales and profits.

Objective 4: Improve cost management to ensure profitability remains strong, especially during moderate sales months.



Bar chart, effectively comparing two quantitative variables (sales and profit) across different categories (months).

Axes:
X-axis: Months
Y-axis (left): Sales in USD
Y-axis (right): Profit in USD 

#Sales Analysis

1. Highest Sales: March recorded the highest sales among all the months.
Possible Reasons: This could be due to various factors such as seasonality, promotions, or specific marketing campaigns that positively influenced customer purchasing behavior.

2. Second-Highest Sales: September followed March in terms of sales.
Considerations: Examining any particular events, promotions, or external factors during September might provide insights into the factors contributing to increased sales during this month.

3. High Sales: November is another month with notable sales.
Seasonal Considerations: Given that November is often associated with Black Friday and holiday shopping, the high sales could be attributed to increased consumer spending during this period.

4. Significant Sales: While slightly lower than November, October shows substantial sales.
Pre-Holiday Preparations: Similar to November, October might benefit from pre-holiday shopping as customers prepare for upcoming festivities.

5. Moderate Sales: August is characterized by moderate sales.
Summer Trends: Sales patterns in August might be influenced by summer-related factors, and analyzing customer behavior during this season could offer insights.

6. Considerable Sales: Despite being a holiday month, December's sales are not the highest.
Year-End Factors: Factors like end-of-year financial planning, budget constraints, or specific product offerings could impact December sales.

7. Moderate Sales: June exhibits moderate sales.
Mid-Year Considerations: Analyzing mid-year trends and potential promotions during this period may provide context for June's sales performance.

8. Lower Sales: January has lower sales compared to the months mentioned earlier.
Post-Holiday Effects: Reduced consumer spending post-holidays might contribute to the lower sales in January.
July:

9. Moderate Sales: July's sales are moderate.
Summer Shopping: Similar to August, summer-related trends and potential promotions could influence July sales.

10. Moderate Sales: May follows a moderate sales trend.
Spring/Early Summer Considerations: Seasonal patterns related to spring and early summer could impact customer spending.

11. Lowest Sales: February has the lowest sales among the mentioned months.
Post-Holiday: Post-holiday fatigue and reduced consumer spending could contribute to lower sales in February.

12. Moderate Sales: April exhibits moderate sales.
Spring Considerations: Spring-related trends and potential promotions may impact sales in April.



#Profit Analysis

1. Highest Profit: March stands out with the highest profit among all the months. Potential Correlation: The factors contributing to high sales in March might also be influencing higher profits. Special promotions, increased demand, or effective cost management could be contributing to this peak.

2. Second-Highest Profit: September follows March with the second-highest profit. Similar Drivers: Similar to March, understanding the events or strategies contributing to high sales in September can help explain the observed high profitability.

3. High Profit: November is identified as a month with high profits. Holiday Sales Impact: November's profitability may be influenced by increased sales during the holiday season, potentially leading to higher profit margins.

4. Significant Profit: October demonstrates noteworthy profitability. Pre-Holiday Preparation: As customers prepare for holidays in October, increased sales and effective cost management could be driving higher profits.

5. Moderate Profit: August is characterized by moderate profitability. Seasonal Considerations: Understanding factors related to summer trends, potential cost fluctuations, and sales patterns in August can provide insights into the observed profit levels.

6. Moderate Profit: June exhibits moderate profitability. Mid-Year Performance: Analyzing mid-year performance and potential factors impacting profitability during this period may offer insights.

7. Lower Profit: January has lower profitability. Post-Holiday Effects: Reduced consumer spending and potential clearance sales after the holiday season may contribute to lower profits in January.

8. Moderate Profit: May follows a moderate profitability trend. Spring/Early Summer Dynamics: Seasonal considerations, such as spring promotions, may influence profit levels in May.

9. Lowest Profit: February is identified as the month with the lowest profit. Post-Holiday: Reduced consumer spending post-holidays and potential clearance sales could impact profit margins in February.

10. Moderate Profit: April demonstrates moderate profitability. Spring Considerations: Analyzing potential spring-related trends and promotions can provide context for April's profit levels. The analysis of monthly profit trends can guide strategic decision-making by identifying periods of peak profitability and potential areas for improvement during lower-performing months. Consideration of both sales and profit data is crucial for a holistic understanding of the company's financial performance, allowing for targeted efforts to enhance overall profitability.

# Analysis for Plot 2 (Bar Chart with Sub Category)

This plot shows the category wise relationship of profit and sales with respect to the subcategories. Copiers emerge as the most profitable product category, demonstrating a significant contribution to the company's overall profit margin. This finding suggests that Copiers are not only popular among customers but also yield a high-profit margin, making them a strategic focus for future business strategies.

Accessories and Phones closely follow Copiers in terms of profitability. This information indicates that these product categories are also lucrative and contribute substantially to the company's financial success. A targeted marketing approach and inventory management for Accessories and Phones could potentially further enhance their profitability.

On the other hand, Paper, Appliances, Binders, Chairs, and Furnishings exhibit varying levels of profitability. Analyzing the performance of these categories can provide valuable insights into factors influencing their contribution to the company's overall revenue and profit.

This analysis offers a roadmap for strategic decision-making. Focusing on high-profit categories like Copiers, Accessories, and Phones can be a key driver for sustained financial success. Meanwhile, exploring ways to optimize the performance of lower-profit categories can contribute to a more balanced product portfolio.

In conclusion, understanding the profit dynamics of each product category is pivotal for informed decision-making. This analysis not only sheds light on the current state of affairs but also serves as a foundation for developing targeted strategies that align with Coding is Fun Ltd.'s overall business objectives. Consideration of these insights in your project report will provide stakeholders with a comprehensive understanding of the company's financial landscape and potential areas for growth and improvement.

# Analysis for the Sales Calendar (Mothly Basis)

The Red Block in the calendar shows the maximum amount of sales in the calendar which comprises of march followed by orange, yellow, green and blue blocks. So this is the appropriate way for the sales personnal and marketing personnals to analyze the monthly sales on the basis of colors which is a more engaging way of showing the revenue flow. 

1. Identifying Seasonal Trends:
A sales calendar allows you to observe patterns and trends over specific time periods. For example, it can reveal if there are particular months or seasons when sales consistently increase or decrease. This information is crucial for planning marketing strategies, promotions, and inventory management.

2. Analyzing Sales Performance Over Time:
By using a sales calendar, you can track the performance of your business over different time intervals. This analysis helps identify growth patterns, understand cyclical trends, and assess the impact of external factors such as holidays or special events on sales.

3. Optimizing Inventory Management:
Understanding sales patterns through a calendar helps in optimizing inventory levels. By anticipating high-demand periods, businesses can ensure they have sufficient stock to meet customer needs during peak times while avoiding overstocking during slower periods.

4. Planning Promotions and Marketing Campaigns:
Sales calendars aid in planning targeted promotions and marketing campaigns. For instance, if there's historical data showing increased sales during specific months, businesses can plan promotional events or discounts to capitalize on customer demand during those periods.

5. Budgeting and Financial Planning:
Sales calendars play a crucial role in budgeting and financial planning. By having a clear overview of when sales are expected to peak or dip, businesses can allocate resources more efficiently, plan for expenses, and set realistic revenue targets.

6. Comparing Performance Across Time Periods:
With a sales calendar, you can easily compare the performance of different time periods. This allows for the identification of growth or decline in sales, enabling businesses to adapt strategies and take corrective actions when needed.

7. Forecasting and Predictive Analytics:
Historical sales data from a calendar can be used for forecasting future sales. Predictive analytics models can analyze past patterns to make informed predictions about future sales trends, helping businesses make proactive decisions.

8. Understanding Customer Behavior:
Analyzing sales data over time helps in understanding customer behavior. For example, recognizing whether customers tend to make larger purchases during specific seasons or prefer certain products at particular times can guide inventory and marketing decisions.

# Analysis for Pie Chart


1	Analysis: The West region contributes the highest percentage of sales (34.1%). This indicates that the company has a significant customer base or market presence in this region.

•	Managerial Insights: Managers can focus on maintaining and potentially expanding their market share in the West. Strategies such as targeted marketing campaigns or personalized promotions can be employed to further boost sales.


2	Analysis: The South region has a relatively lower share of sales (16.8%) compared to other regions. This might suggest that there is room for growth or improvement in this area.

•	Managerial Insights: Managers may want to investigate the reasons behind the lower sales in the South. Conducting market research, understanding customer preferences, and optimizing logistics can help in enhancing sales performance in this region.


3	Analysis: The Central region contributes to a significant portion of sales (20.1%). This implies a moderate but crucial presence in the market.

•	Managerial Insights: Managers can focus on strategies to maintain and potentially increase the market share in the Central region. Exploring partnerships, improving supply chain efficiency, and addressing customer needs can contribute to sustained growth.


4	Analysis: The East region holds the second-highest sales percentage (29.1%). This suggests a substantial market presence, making it a key area for the company's overall performance.

•	Managerial Insights: Managers can consider reinforcing successful strategies in the East region. They may also explore opportunities for expansion or innovation to capitalize on the existing market share and potentially increase it further.


# Sunburst Chart Analysis: Shipping Region and Shipping Mode Preferences

Standard Class Dominance:

•	Across all regions (West, East, Central, South), the majority of shipments are using the "Standard Class" mode of delivery.
•	Implication: Standard class is the most preferred and widely used shipping mode, likely due to its balance between cost and delivery time.

Second Class Following Standard Class:

•	The second most popular choice for shipping mode is "Second Class" in all regions.
•	Implication: Customers may opt for second-class delivery when they want a balance between cost and speed, and it serves as a common alternative to standard class.

First Class Preference:

•	"First Class" is the third most chosen mode of delivery in the preferences.
•	Implication: First-class delivery is likely selected by customers who prioritize faster shipping and are willing to pay a premium for expedited service.

Same Day Delivery as the Least Preferred:

•	"Same Day Delivery" is the least preferred mode of delivery across all regions.
•	Implication: Same day delivery tends to be the least chosen option, possibly due to higher associated costs or the nature of the products being shipped. It's likely that customers generally prioritize cost-effective and slightly slower delivery options.

Overall Interpretation:

•	The sunburst chart indicates that customers across different regions tend to favor cost-effective and reasonably paced shipping options, with a preference for the "Standard Class" mode. While there is a demand for faster deliveries, it is not as prevalent as the standard and second-class options. Understanding these preferences can help businesses tailor their logistics and shipping strategies to better align with customer expectations, potentially optimizing costs and customer satisfaction.

•	This analysis provides valuable insights into the dynamics of shipping mode preferences, enabling businesses to make informed decisions regarding their logistics and delivery services.


# Analysis of Interactive Box Plot

Average Discount and Profit: The average discount varies across regions, with the East region having the highest average discount (20%) and the West region having the lowest (5%). Similarly, the average profit also varies, with the East region having the highest average profit (15%) and the West region having the lowest (3%).

Correlation between Discount and Profit: There is no correlation between discount and profit for any of the regions. This means that changes in discount do not appear to be associated with changes in profit, at least based on the data provided.

Potential outliers: There is one potential outlier, which is a data point in the East region with a discount of 20% and a profit of 15%. This data point falls outside the typical range of values for both discount and profit in the East region.

# Analysis of Scatter Plot (Profit V/s Discounts)

1. Total quantity of seats plus discounts is higher than total quantity of seats plus profit for all regions. This suggests that the discounts offered are reducing the overall revenue generated from seat sales.

2. The East region has the highest total quantity of seats plus discounts and the lowest total quantity of seats plus profit. This indicates that the East region is offering the most discounts but also has the lowest overall revenue.

3. The West region has the lowest total quantity of seats plus discounts and the highest total quantity of seats plus profit. This suggests that the West region is offering the fewest discounts and has the highest overall revenue.

4. The Central region has values in between the East and West regions for both total quantity of seats plus discounts and total quantity of seats plus profit. This suggests that the Central region is offering moderate discounts and has moderate overall revenue.

Explanations for the Analysis

1. Regional Differences in Pricing Strategy: The observed differences in discounts and profits between regions could be due to varying pricing strategies. The East region might be focusing on attracting customers with lower prices and higher discounts, while the West region might be targeting a more premium market with higher prices and fewer discounts.

2. Demand and Competition: Regional variations in demand and competition could also play a role. If there is higher demand or less competition in the West region, businesses might be able to charge higher prices without offering significant discounts. Conversely, if there is lower demand or more competition in the East region, businesses might need to offer discounts to attract customers.

3. Other Factors: Other factors, such as the types of seats offered, amenities, and target customer segments, could also influence the pricing and discounting strategies in different regions.